docs/source/peer_ledger_snapshot.md

```diff
@@ -0,0 +1,154 @@
+
+Before taking a snapshot, it is a best practice to confirm the current ledger height by issuing a command similar to:
+
+```
+peer channel getinfo -c <name of channel>
+```
+
+You will see a response similar to:
+
+```
+Blockchain info: {"height":970,"currentBlockHash":"JgK9lcaPUNmFb5Mp1qe1SVMsx3o/22Ct4+n5tejcXCw=","previousBlockHash":"f8lZXoAn3gF86zrFq7L1DzW2aKuabH9Ow6SIE5Y04a4="}
+```
+
+In this example, the ledger height is `970`.
+
+A snapshot request can be submitted by issuing a command similar to:
+
+```
+peer snapshot submitrequest -C <name of channel> -b <ledger height where snapshot will be taken> --peerAddress <address of peer> --tlsRootCertFile <path to root certificate of the TLS CA>
+```
+
+For example:
+
+```
+peer snapshot submitrequest -C testchannel -b 970 --peerAddress 127.0.0.1:22509 --tlsRootCertFile tls/cert.pem
+```
+
+**If you give a ledger height of `0`, the snapshot is taken immediately. This is useful for cases when an organization is interested in generating a snapshot that will be used by one of its own peers and does not intend to share the data with another organization. Do not take these "immediate" snapshots in cases when snapshots will be evaluated from multiple peers, as it increases the likelihood that the snapshots will be taken at different ledger heights.**
+
+If the request is successful, you will see a `Snapshot request submitted successfully` message.
+
+You can list the pending snapshots by issuing a command similar to:
+
+```
+peer snapshot listpending -C testchannel --peerAddress 127.0.0.1:22509 --tlsRootCertFile tls/cert.pem
+```
+
+You will see a response similar to:
+
+```
+Successfully got pending snapshot requests [970]
+```
+
+When a snapshot has been generated for a particular block height, the pending request for that block height will no longer appear in the list. You can also verify that a snapshot has been create successfully by looking at the peer logs or by checking the expected location of the snapshot, `{snapshotsFolder}/completed/{channelName}/{lastBlockNumberInSnapshot}`, for the snapshot.
+
+To delete a snapshot request, simply exchange `submitrequest` with `cancelrequest`. For example:
+
+```
+peer snapshot cancelrequest -C testchannel -b 1000 --peerAddress 127.0.0.1:22509 --tlsRootCertFile tls/cert.pem
+```
+
+If you submit the `listpending` command again, the snapshot should no longer appear.
+
+### Contents of a snapshot
+
+Snapshots are stored on a peer in a snapshot folder that is further organized by channel name and the block number when the snapshot was taken: `{snapshotsFolder}/completed/{channelName}/{lastBlockNumberInSnapshot}`. Once the peer generates a snapshot in this folder, the peer does not use that folder for any purpose and it is safe to transfer or compress the snapshot using external tools.
+
+As mentioned above, the snapshot contains four folders:
+
```

Comment:
Snapshot is a single folder that contains files for different data items listed below.